File: Modular_Arithmetic/ModularInverse.py
def modular_inverse(a, m):
    # Brute-force
    # a * a^-1 mod m == 1 
    for inv in range(0, m):
        if (a*inv)%m == 1:
            return inv
    print('There is no modular inverse')

if __name__ == '__main__':
    print(modular_inverse(9, 31))

File: Modular_Arithmetic/ModularInverse.java
package RSA;

public class ModularInverse {
    // Brute-force
    // a * a^-1 mod m == 1 

    public int find_inverse(int a, int m){
        for(int inv = 0; inv<m; inv++){
            if((a*inv)%m == 1){
                return inv;
            }
        }
        return -1;
    }
}
